requirements/EVENT_SERVICES_AURORA_0001/register-event-listeners/work/eventlistener2/C/src/eventlistener2.c

```diff
@@ -0,0 +1,72 @@
+/* Body file for function EventListener2
+ * Generated by TASTE (kazoo/templates/skeletons/c-body/function.tmplt)
+ * You can edit this file, it will not be overwritten
+
+    !! IMPORTANT                                                        !!
+    !! When you modify your design (interface view), you must update    !!
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "eventlistener2.h"
+#include <stdio.h>
+
+#define SEC_TO_NS(sec) ((sec)*1000000000)
+
+void eventlistener2_startup(void)
+{
+    asn1SccT_UInt32 eventId = 3;
+    bool shouldSubscribe = true;
+    eventlistener2_RI_subscribe_to_event(&eventId, &shouldSubscribe);
+    eventId = 2;
+    eventlistener2_RI_subscribe_to_event(&eventId, &shouldSubscribe);
+}
+
+void eventlistener2_PI_receive_event
+      (const asn1SccT_EventMessage *IN_eventmessage)
+
+{
+    uint64_t time = 0;
+    uint32_t seconds = 0;
+    uint32_t nanoseconds = 0;
+
+    seconds += (uint32_t)IN_eventmessage->timestamp.c1 << 24;
+    seconds += (uint32_t)IN_eventmessage->timestamp.c2 << 16;
+    seconds += (uint32_t)IN_eventmessage->timestamp.c3 << 8;
+    seconds += (uint32_t)IN_eventmessage->timestamp.c4;
+
+    nanoseconds += (uint32_t)IN_eventmessage->timestamp.f1 << 24;
+    nanoseconds += (uint32_t)IN_eventmessage->timestamp.f2 << 16;
+    nanoseconds += (uint32_t)IN_eventmessage->timestamp.f3 << 8;
+
+    time = SEC_TO_NS((uint64_t)seconds) + (uint64_t)nanoseconds;
+
+    printf("Event received by Listener 2:\n");
+    printf("  event id =       %ld\n", IN_eventmessage->id);
+    switch(IN_eventmessage->eventType)
```

Comment:
Normally a bit of code duplication in test models is not a problem, but have you considered providing a lib/header for handling the timestamps and event printing?

Reply:
Done, added libs